server/operator.go

```diff
@@ -0,0 +1,201 @@
+// Copyright 2016 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package server
+
+import (
+	"github.com/juju/errors"
+	"github.com/ngaut/log"
+	"github.com/pingcap/kvproto/pkg/metapb"
+	"github.com/pingcap/kvproto/pkg/pdpb"
+	"github.com/pingcap/kvproto/pkg/raftpb"
+)
+
+// Operator is the interface to do some operations.
+type Operator interface {
+	// Do does the operator, if finished then return true.
+	Do(region *metapb.Region, leader *metapb.Peer) (bool, *pdpb.RegionHeartbeatResponse, error)
+	// Check checks whether the operator has been finished.
+	Check(region *metapb.Region, leader *metapb.Peer) (bool, error)
+}
```

Comment:
the raft can `AddPeer` successfully even the store doesn't exist...
Like @qiuyesuifeng said, we will support Cancel mechanism later. 
E.g, if the ops run not ok for a long time or many times, we can cancel it.